pkg/virt-operator/webhooks/kubevirt-update-admitter.go

```diff
@@ -162,3 +176,116 @@ func validateCertificates(certConfig *v1.KubeVirtSelfSignConfiguration) []metav1
 
 	return statuses
 }
+
+//validateWorkloadPlacement
+func validateWorkloadPlacement(placementConfig *v1.NodePlacement, client kubecli.KubevirtClient) []metav1.StatusCause {
+	statuses := []metav1.StatusCause{}
+
+	const (
+		dsName    = "placement-validation-webhook"
+		namespace = "default"
+		mockLabel = "kubevirt.io/choose-me"
+		podName   = "placement-verification-pod"
+		mockUrl   = "test.only:latest"
+	)
+
+	mockDaemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: dsName,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					mockLabel: "",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: podName,
+					Labels: map[string]string{
+						mockLabel: "",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  podName,
+							Image: mockUrl,
+						},
+					},
+					// Inject placement fields here
+					NodeSelector: placementConfig.NodeSelector,
+					Affinity:     placementConfig.Affinity,
+					Tolerations:  placementConfig.Tolerations,
+				},
+			},
+		},
+	}
+
+	_, err := client.AppsV1().DaemonSets(namespace).Create(context.Background(), mockDaemonSet, metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}})
+
+	if err != nil {
+		statuses = append(statuses, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: err.Error(),
+		})
+	}
+	return statuses
+}
+
+func validateInfraPlacement(placementConfig *v1.NodePlacement, client kubecli.KubevirtClient) []metav1.StatusCause {
```

Comment:
@zcahana You are correct about it being conceptually similar, the potential issue that may arise here is a security/RBAC issue - in this manner the virt-operator may be restricted to create only the resource types that appear in it's install strategy. Therefore it may be forbidden for it to create a single pod. So we may fail here even if we try only to do a dry-run

Reply:
That's a great point. Looking at the manifests, the virt-operator is indeed limited to creating deployments/daemonsets but can't create pods. I take it that the webhook runs in the same pod as virt-operator, or at least with the same service account. It would be interesting to test whether this also affects dry-run, but my guess is that it does.

Do you think this usage here merits adding a CREATE pods permission to the webhook pod?

Reply:
@zcahana I prefer the least privilege approach in that case. Also confirmed it with Roman